modules/imgcodecs/src/grfmt_avif.cpp

```diff
@@ -358,13 +367,17 @@ bool AvifEncoder::writeToOutput(const std::vector<Mat> &img_vec,
 
     images.emplace_back(ConvertToAvif(img, do_lossless, bit_depth));
   }
-  for (const AvifImageUniquePtr &image : images) {
+
+  int timestamp = 0;
+  for (size_t i = 0; i < images.size(); i++)
+  {
     OPENCV_AVIF_CHECK_STATUS(
-        avifEncoderAddImage(encoder_, image.get(), /*durationInTimescale=*/1,
-                            flag),
+        avifEncoderAddImage(encoder_, images[i].get(), timestamp, flag),
         encoder_);
+    timestamp += animation.durations[i];
```

Comment:
move this line to above avifEncoderAddImage because avifEncoderAddImage expects a duration, not a timestamp.